ledger/simulation/simulator.go

```diff
@@ -85,13 +87,15 @@ type EvalFailureError struct {
 
 // Simulator is a transaction group simulator for the block evaluator.
 type Simulator struct {
-	ledger simulatorLedger
+	ledger     simulatorLedger
+	nodeConfig config.Local
 }
 
 // MakeSimulator creates a new simulator from a ledger.
-func MakeSimulator(ledger *data.Ledger) *Simulator {
+func MakeSimulator(ledger *data.Ledger, nodeConfig config.Local) *Simulator {
```

Comment:
I do not think we should pass in the `config.Local`. Instead, just the values that we care about, which allows the argument names to provide a kind of documentation about what is going into `MakeSimulator`. This seems like just handing in a big black box of stuff.

Reply:
Would having a `Config` struct in the simulator package with only the values it needs sound better to you? Or is there a different idea you're thinking of?

Reply:
This is actually mirroring #5366, see thread https://github.com/algorand/go-algorand/pull/5366#discussion_r1197936697

Me and Jason are both thinking about providing ability to disabling these features (pc, unlimited-resource) over local node config, so we are passing in the `config.Local` on `MakeSimulator`.

If we are going down your line, would it be a concern to have too many args for `MakeSimulator`?

Reply:
I think maybe I'm trying to make it hard for us to keep putting more and more flags into `config.Local`. I want us to feel the pain every time we add another flag to control some little aspect of the node, because I think we should exercise restraint, and try to put related things under one umbrella, as we do for "dev api" stuff.

So I'm thinking that `MakeSimulator()` should have to take, a boolean for each thing we're flagging, so we are confronted by the complexity we're introducing.

All that being said, I don't feel _incredibly_ strong here. It just feels wrong to pass this thing around, because it seems like we're essentially making the config a big pile of global variables.

Reply:
ok, now I understood. The thinking is not tossing around a huge chunk of `config.Local` variable, and `MakeSimulator` should only consume a small subset of the information from `config.Local`, rather than using all the variable instance.

I am on board with that. See [avoid passing config.Local whole thing around](https://github.com/algorand/go-algorand/pull/5360/commits/c213a6b6b1e3d3bb2a0e5ce16348a14ddeeac06f) for the change.